### ISSUE TITLE
polygon.editing.enable() error

### ISSUE THREAD

**Issue by wmaiouiru** [Leaflet/Leaflet.draw#804]
- [x] I'm reporting a bug, not asking for help
- [x] I've looked at the [documentation](http://leaflet.github.io/Leaflet.draw/dist/reference-0.4.1.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet Draw code issue, not an issue with my own code nor with the framework, I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported

----

## How to reproduce

- Leaflet version I'm using: 1.0.3
- Leaflet Draw version I'm using: 0.4.9 and tried 0.4.13
- Browser (with version) I'm using:  Chrome 62.0.3202.94
- OS/Platform (with version) I'm using: macOS Sierra Version 10.12.4
- step 1
The following example code are broken. 
http://leaflet.github.io/Leaflet.draw/docs/examples/snapping.html
http://leaflet.github.io/Leaflet.draw/docs/examples/edithandlers.html
- step 2

## What behaviour I'm expecting and which behaviour I'm seeing
The problem seems to be at:
polygon.editing.enable()
In L.Edit.PolyVerticesEdit, editing in poly.options.editing.className is not defined. 
## Minimal example reproducing the issue

- [x] this example is as simple as possible
- [x] this example does not rely on any third party code

Using [jsfiddle](https://jsfiddle.net/ddproxy/mvp7hgou/) or another example site.
http://leaflet.github.io/Leaflet.draw/docs/examples/snapping.html
http://leaflet.github.io/Leaflet.draw/docs/examples/edithandlers.html
Check the console output.

**Comment by Wh1teFr0st** [comment]
In my case it's the same problem but even sometimes for all shapes like lines, circle etc. in Chrome:

Uncaught TypeError: Cannot read property 'className' of undefined
    at e.addHooks (leaflet.draw.js:8)
    at leaflet.draw.js:8
    at e._eachVertexHandler (leaflet.draw.js:8)
    at e.addHooks (leaflet.draw.js:8)
    at enable (leaflet.js:5)
    at e._enableLayerEdit (leaflet.draw.js:9)
    at eachLayer (leaflet.js:5)
    at e.addHooks (leaflet.draw.js:9)
    at enable (leaflet.js:5)
    at enable (leaflet.draw.js:9)

I'm using Leaflet 1.2.0 and Leaflet.draw 0.4.13. This version have less "Deprecated include of L.Mixin.Events:" errors by the way. So i also need to fall back to version 0.4.12 at the moment. 

Error occurs on Chrome 62.0.3202.94 (64-Bit) and Opera 49.0.2725.34 (PGO). Firefox Quantum 57.0 (64-Bit) can handle 0.4.13.

**Comment by pilarArr** [comment]
I have this issue too, also worked in 0.4.12.

I get this error when I activate editing on a GeoJSON layer with a LineString, which I guess would translate to a polyline

**Comment by xinaris** [comment]
I solved it for now with this:
polygon.options.editing || (polygon.options.editing = {});
polygon.editing.enable();

from
#364

**Comment by TPABHuKOB** [comment]
Pls fix it in libary

**Comment by errazkisibawaih** [comment]
I had the same issue, but with @NovinNovin code changes the issue is fixed.
Please fix it the lib.

**Comment by romelperezjs** [comment]
Yes please. Can anyone fix it in the library? with @novinnovin PR?

**Comment by cecilepolestarglobal** [comment]
This error is still happening, is there an open branch for a fix ?

**Comment by knownasilya** [comment]
I think #838 closes this issue?

**Comment by Chris1234567899** [comment]
The same thing seems to happen to other geometry types as it seems:

marker -> different error (Cannot read property 'enable' of undefined)
circle -> works
linestring -> same error
rectangle -> works
polygon -> same error

**Comment by akaivola** [comment]
Workaround:

You can put placeholder `editing` property in shapeOptions:

```
new L.Control.Draw(
  {draw: {
    polygon: {
      shapeOptions: {
        editing: {
          className: ""}}}}}})
```

**Comment by pilarArr** [comment]
The leaflet draw examples for v1+ actually use v0.4.2

**Comment by geuning2** [comment]
In my case
at Edit,Poly.js,
replace all this._ defaultShape() with this._poly._defaultShape()

**Comment by alexandervlpl** [comment]
Sadly this library seems to be abandoned now. I ran into this and similar issues while upgrading Leaflet and Draw, now considering migrating to [Geoman](https://geoman.io/leaflet-geoman) instead. :cry:
